sdk/storage/Azure.Storage.Queues/src/QueueClient.cs

```diff
@@ -2509,6 +2763,27 @@ public virtual Uri GenerateSasUri(QueueSasPermissions permissions, DateTimeOffse
             return sasUri.ToUri();
         }
         #endregion
+
+        #region Encoding
+        private static BinaryData ToBinaryData(string input)
+        {
+            if (input == null)
+            {
+                return null;
+            } else
+            {
+                return new BinaryData(input);
+            }
+        }
+
+        private void AssertEncodingForEncryption()
+        {
+            if (UsingClientSideEncryption && _messageEncoding != QueueMessageEncoding.UTF8)
+            {
+                throw new ArgumentException($"Message encoding can't be customized if encryption-enabled QueueClient is used.");
```

Comment:
"Can't be customized" is confusing and assumes customers understand all the defaults.  Can we say something more like `$"{nameof(SpecializedQueueClientOptions)} requires {nameof(QueueMessageEncoding)}.{nameof(QueueMessageEncoding.UTF8)} if {nameof(SpecializedQueueClientOptions.ClientSideEncryption)} is enabled."`

Reply:
Also why can't we encode + encrypt?  Wouldn't it just be symmetric transformations before/after encrypting?

Reply:
Encryption implicitly involve base64 encoding, so we want to prevent double base64 encoding.